src/app/services/study-set-dev.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { StudySetService } from './study-set.service';
+import { StudySetData, StudySetModel } from '../data-models/studyset-model';
+import { FlashcardData, FlashcardModel } from '../data-models/flashcard-model';
+import { SequenceData, SequenceModel } from '../data-models/sequence-model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StudySetDevService extends StudySetService{
+  readonly baseUrl = 'http://localhost:3000/';
+  readonly studySetEndpoint = "studySets";
+  readonly serverStateEndpoint = "state";
+
+  override async getStudySet(id: string): Promise<StudySetData> {
+    const studySet = await fetch(`${this.baseUrl}${this.studySetEndpoint}/${id}`)
```

Comment:
I'm not sure what scenario this would happen in, but this function stops unexpectedly when an id that is not in the db is requested. You might not need to fix this since this is a dev service.

Reply:
Yeah, ideally we create an additional error handler class that deals with what to do is the response is empty (i.e. there is no study set with the id on the server). I was not sure how we wanted to handle that across that app because there were several approaches that could be taken, such as show a error page (like "404 not found") or show an empty set. I didn't want to return an undefined or empty set because that would mask the error and make future debugging a pain.

Eventually, the study set id's to fetched will be determined by other services for getting the id's 1) user's recently studied 2) user's owned sets, and 3) sets based on a search query. 

Those can become new story points because this current task was already really large. Sorry I didn't mention this when you asked at the meeting yesterday.